### PR TITLE
Implement edit/delete, add test coverage, refactor notifications

### LINKED ISSUE #464: Handle Reject option
Right now, we don't handle the reject option on the admin. We should check on the approved param and either approve the change, or reject it. Also, there is a typo on the accept button name in the form.

### PR DESCRIPTION
**Pull request by doomspork**
This PR expands on #462 by @gemantzu to take a slightly different approach, refactor the notifications, and to add test coverage.

Closes #464 #458

### CODE REVIEW

**Comment by doomspork** [comment]
This is a start on the test coverage. We still need to create tests for the web portion and schema (if we decide that's warranted)

File: lib/notify/slack.ex [hunk 1]
```diff
@@ -1,4 +1,4 @@
-defmodule Companies.Slack do
+defmodule Notify.Slack do
```
Comment (doomspork) [inline lib/notify/slack.ex]: I'll add `bypass` to the project to test this 👍

**Comment by doomspork** [comment]
While adding test coverage I dropped all of the suffixes from the module functions (ie `Jobs.create_job` -> `Jobs.create`)

File: lib/companies_web/controllers/job_controller.ex [hunk 2]
```diff
@@ -5,19 +5,19 @@ defmodule CompaniesWeb.JobController do
...
-    company = Companies.get_company!(company_id)
+    changeset = Jobs.change(%Job{company_id: company_id})
+    company = Companies.get!(company_id)
     render(conn, "new.html", changeset: changeset, company: company)
   end
 
   def create(conn, %{"job" => params}) do
     case Jobs.create(params, current_user(conn)) do
-      {:ok, _job} ->
+      {:ok} ->
```
Comment (gemantzu) [inline lib/companies_web/controllers/job_controller.ex]: On successful creation, what gets returned is {:ok, %PendingChange{}}, which makes this fail to pattern match.

File: lib/companies/pending_changes.ex [hunk 3]
```diff
@@ -5,26 +5,33 @@ defmodule Companies.PendingChanges do
...
-  alias Companies.{Repo, Slack}
+  alias Companies.Repo
   alias Companies.Schema.{Company, Job, Industry, PendingChange}
 
   def all do
     query =
       from p in PendingChange,
-        select: p,
-        where: p.approved == false
+        where: is_nil(p.approved)
```
Comment (gemantzu) [inline lib/companies/pending_changes.ex]: You need to add a migration to make them not default to false in db for this to work.

File: test/companies/pending_changes_test.exs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      assert {:ok, %{approved: true}} = PendingChanges.approve(id, true)
+      assert %{total_entries: 0} = Companies.all()
+    end
+
+    test "rejects a pending change" do
+      %{total_entries: pre_count} = Companies.all()
+
+      %{id: id} = insert(:pending_change)
+      assert {:ok, %{approved: false}} = PendingChanges.approve(id, false)
+      assert 0 == length(PendingChanges.all())
```
Comment (tajchumber) [inline test/companies/pending_changes_test.exs]: use `Enum.empty?/1` intead of `0 == length(PendingChanges.all())`

File: test/companies/pending_changes_test.exs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    end
+  end
+
+  describe "approve/2" do
+    test "approves a pending create change" do
+      %{total_entries: pre_count} = Companies.all()
+
+      %{id: id} = insert(:pending_change)
+      assert {:ok, %{approved: true}} = PendingChanges.approve(id, true)
+      assert 0 == length(PendingChanges.all())
```
Comment (tajchumber) [inline test/companies/pending_changes_test.exs]: use `Enum.empty?/1` intead of `0 == length(PendingChanges.all())`

File: test/companies/pending_changes_test.exs (hunk 4 above)
Comment (doomspork) [reply test/companies/pending_changes_test.exs]: Done!

**Pull request merged**